pages/borrow/[symbol].tsx

```diff
@@ -61,13 +65,146 @@ const Borrow: NextPage<
     () => parseUnits(collateralValue, collateral?.decimals ?? 18),
     [collateralValue, collateral?.decimals],
   )
+  const collateralBalance = useMemo(
+    () => (collateral ? balances[collateral.address] : 0n),
+    [balances, collateral],
+  )
 
   const loanAmount = useMemo(
     () => parseUnits(loanValue, asset.underlying.decimals),
     [loanValue, asset.underlying.decimals],
   )
 
-  console.log(collateralAmount, loanAmount)
+  const maxLiquidationTargetLtv = useMemo(
+    () =>
+      collateral
+        ? BigInt(
+            asset.collaterals.find(({ underlying }) =>
+              isAddressEqual(underlying.address, collateral.address),
+            )?.liquidationTargetLtv || 0n,
+          )
+        : 0n,
+    [asset.collaterals, collateral],
+  )
+
+  const maxLoanAmountExcludingCouponFee = useMemo(() => {
+    if (epochs === 0) {
```

Comment:
`if (epochs === 0 || !collateral || !asset) {`

This will simplify the code below as the null checks can be omitted.

Reply:
https://github.com/clober-dex/coupon.finance/pull/74/commits/7c737de08cd2499e84cd71e9d1f845c38a93cff0